src/braket/circuits/synthesis/one_qubit_decomposition.py

```diff
@@ -0,0 +1,211 @@
+            of the rotation.
+        rotation_matrix (np.ndarray): the matrix representation
+            of the rotation.
+    """
+
+    def __init__(self, U: np.ndarray, atol: float = 1e-8, rtol: float = 1e-5):
+
+        self.atol = atol
+        self.rtol = rtol
+
+        if U.shape != (2, 2):
+            raise ValueError("Input matrix has to be 2x2.")
+
+        is_unitary(U, atol=self.atol, rtol=self.rtol, raise_exception=True)
+
+        self.U = U
+        su = to_su(self.U)
+        self.phase = complex(np.linalg.det(U)) ** 0.5
+
+        # Calculate zyz Euler angles.
+        weight_matrix = np.array([[1, 1], [-1, 1]])
+
+        r1 = 2 * np.arctan2(abs(su[1, 0]), abs(su[0, 0]))
+
+        r0r2 = np.angle(su[1, :])
+
+        # have to check if r2 is 0 to avoid gimbal lock.
+        if abs(su[0, 1]) < self.atol:
+            r0 = 2 * np.angle(su[1, 1])
+            r2 = 0
+        else:
+            r0r2 = weight_matrix @ np.angle(su[1, :])
+            r0, r2 = np.where(r0r2 < 0, r0r2 + 2 * np.pi, r0r2)
+
+        self._euler = np.array([r0, r1, r2])
+
+        # Calculate the quaternion representation.
+
+        q0 = np.real(su[0, 0])
+        q1 = -np.imag(su[0, 1])
+        q2 = -np.real(su[0, 1])
+        q3 = -np.imag(su[0, 0])
+
+        self._quat = np.array([q0, q1, q2, q3])
+
+    def euler_angles(self, axes: str = "zyz"):
+        """
+        Find the Euler angle of 1-qubit gates.
+
+        Default is the following zyz decomposition:
+
+            SU(2) = exp(i * phase) *
+
+                    [[exp(-1j*r1), 0],
+                     [0,  exp(1j*r1)]] @
+
+                    [[cos(r2), -sin(r2)],
+                     [sin(r2),  cos(r2)]] @
+
+                    [[exp(-1j*r3), 0],
+                     [0,  exp(1j*r3)]]
+
+        Args:
+            axes (str): the decomposed axes.
+
+        Raises:
+            NotImplementedError: if the decomposition method is not
+                implemented.
+        """
+
+        if axes == "zyz":
+            return self._euler
+
+        elif axes == "zxz":
+            r0, r1, r2 = self._euler
+            return np.array([r0 + 0.5 * np.pi, r1, r2 - 0.5 * np.pi])
+
+        else:
+            raise NotImplementedError("The decomposition method is not implemented.")
+
+    @property
+    def quaternion(self):
+        """
+        Quaternion representation of 1-q decomposition.
+        """
+
+        return self._quat
+
+    @property
+    def canonical_vector(self):
+        """
+        Return the rotation axis for the
+        axis decomposition of 1-qubit gates.
+
+        SU(2) = exp(-0.5j * theta * (xX + yY + zZ))
+
+        To align with 2-qubit gate decomposition, we name it
+        the canonical_vector.
+        """
+        norm = np.linalg.norm(self._quat[1:])
+
+        if norm < self.atol:
+            return np.array([1, 0, 0])
+
+        return self._quat[1:] / norm
+
+    @property
+    def rotation_angle(self):
```

Comment:
Don't see where this is used other than the `__repr__`; if this is the only usage, remove the method altogether.